src/components/Files/CameraCaptureDialog.tsx

```diff
@@ -24,27 +30,43 @@ export interface CameraCaptureDialogProps {
   setPreview?: (isPreview: boolean) => void;
 }
 
+const lastUsedCameraDeviceIdAtom = atomWithStorage<string | null>(
+  "last_used_camera_device_id",
+  null,
+);
+
 export default function CameraCaptureDialog(props: CameraCaptureDialogProps) {
   const { t } = useTranslation();
 
   const { open, onOpenChange, onCapture, onResetCapture, setPreview } = props;
   const isLaptopScreen = useBreakpoints({ lg: true, default: false });
-  const [cameraFacingMode, setCameraFacingMode] = useState(
-    isLaptopScreen ? "user" : "environment",
-  );
+  const [cameraFacingMode, setCameraFacingMode] = useState("environment");
   const [previewImage, setPreviewImage] = useState<string | null>(null);
+  const [showCameraSelector, setShowCameraSelector] = useState(false);
+  const [selectedDeviceId, setSelectedDeviceId] = useAtom(
+    lastUsedCameraDeviceIdAtom,
+  );
   const webRef = useRef<Webcam>(null);
 
-  const videoConstraints = {
-    width: { ideal: 4096 },
-    height: { ideal: 2160 },
-    facingMode: cameraFacingMode,
-  };
+  const videoConstraints =
+    isLaptopScreen && selectedDeviceId
+      ? { deviceId: selectedDeviceId }
+      : { facingMode: cameraFacingMode };
 
-  const { startStream, stopStream } = useMediaStream({
-    constraints: { video: { facingMode: cameraFacingMode } },
+  const { startStream, stopStream, devices, state } = useMediaStream({
```

Comment:
consider having clear names for variables when being returned.

```suggestion
  const { startStream, stopStream, devices, cameraPermission } = useMediaStream({
```